plugins/check_pgsql.c

```diff
@@ -344,10 +343,7 @@ process_arguments (int argc, char **argv)
 				pgport = optarg;
 			break;
 		case 'd':     /* database name */
-			if (!is_pg_dbname (optarg)) /* checks length and valid chars */
-				usage2 (_("Database name is not valid"), optarg);
-			else /* we know length, and know optarg is terminated, so us strcpy */
-				strcpy (dbName, optarg);
+			strcpy (dbName, optarg);
```

Comment:
The ``is_pg_dbname`` check, also checked whether ``optarg`` is at most ``NAMEDATALEN`` octets long.
Using ``strcpy(2)`` without knowing that src string is not longer than the destination buffer is unsafe.